azure-ipam/ipam.go

```diff
@@ -85,17 +106,25 @@ func (p *IPAMPlugin) CmdAdd(args *cniSkel.CmdArgs) error {
 		p.logger.Error("Failed to interpret CNS IPConfigResponse", zap.Error(err), zap.Any("response", resp))
 		return cniTypes.NewError(ErrProcessIPConfigResponse, err.Error(), "failed to interpret CNS IPConfigResponse")
 	}
-	p.logger.Debug("Parsed pod IP", zap.String("podIPNet", podIPNet.String()))
-
-	cniResult := &types100.Result{
-		IPs: []*types100.IPConfig{
-			{
-				Address: net.IPNet{
-					IP:   net.ParseIP(podIPNet.Addr().String()),
-					Mask: net.CIDRMask(podIPNet.Bits(), 32), // nolint
-				},
-			},
-		},
+	for _, IPNet := range *podIPNet {
+		p.logger.Debug("Parsed pod IP", zap.String("podIPNet", IPNet.String()))
+	}
+
+	cniResult := &types100.Result{}
+	for _, IPNet := range *podIPNet {
+		ipConfig := &types100.IPConfig{}
+		if net.ParseIP(IPNet.Addr().String()).To4() != nil {
```

Comment:
I would probably use `if IPNet.Addr().Is4() {` instead of parsing and checking for `nil`

Reply:
updated the if statement